src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java

```diff
@@ -794,14 +741,96 @@ public void updateElement(final int index, @NotNull final Pane rowPane)
         });
     }
 
+    /**
+     * Update the display for the happiness
+     */
+    private void updateHappiness()
+    {
+        final HappinessData happiness = ColonyManager.getColony(building.getColony().getID()).getCitizenManager().getHappiness();
+        for (final String str : new String[] {GUARD_HAPPINESS_LEVEL, HOUSE_HAPPINESS_LEVEL, SATURATION_HAPPINESS_LEVEL})
+        {
+            final Image image = findPaneOfTypeByID(str, Image.class);
+            switch (happiness.getGuards())
```

Comment:
I am pretty sure this is wrong. You are querying for all three happiness levels the same value.